crates/katana/core/src/service/messaging/ethereum.rs

```diff
@@ -173,45 +173,42 @@ impl Messenger for EthereumMessaging {
             return Ok(vec![]);
         }
 
-        let starknet_messaging = StarknetMessagingLocal::new(
-            self.messaging_contract_address,
-            self.provider_signer.clone(),
-        );
+        let starknet_messaging =
+            StarknetMessagingLocal::new(self.messaging_contract_address, self.provider.clone());
 
         let hashes = parse_messages(messages);
 
         debug!("Sending transaction on L1 to register messages...");
-        match starknet_messaging
-            .add_message_hashes_from_l2(hashes.clone())
+
+        let receipt = starknet_messaging
+            .addMessageHashesFromL2(hashes.clone())
```

Comment:
Is there a way to have this method in `snake_case` as mentioned in the other comment?

Reply:
I've tried to put a snake case renaming constraint in the `sol!` macro but it doesn't seem to work. The functions around this macro are not yet all stable as specified in their repo and so I tell myself that it is surely related.